src/encoder.rs

```diff
@@ -21,6 +24,80 @@ fn encode_vlq_diff(out: &mut String, a: u32, b: u32) {
     encode_vlq(out, i64::from(a) - i64::from(b))
 }
 
+fn encode_rmi(out: &mut Vec<u8>, indices: &[usize]) {
+    fn encode_byte(b: u8) -> u8 {
+        match b {
+            0..=25 => b + b'A',
+            26..=51 => b + b'a' - 26,
+            52..=61 => b + b'0' - 52,
+            62 => b'+',
+            63 => b'/',
+            _ => panic!("invalid byte"),
+        }
+    }
+
+    let mut data = [0u8; 8];
+
+    let bits = data.view_bits_mut::<Lsb0>();
+    for &i in indices {
+        bits.set(i, true);
+    }
+
+    // trim zero at the end
+
+    let mut last = 0;
+    for (idx, bit) in bits.iter().enumerate() {
+        if *bit {
+            last = idx;
+        }
+    }
+    let bits = &mut bits[..last + 1];
+
+    for byte in bits.chunks(6) {
+        let byte = byte.load::<u8>();
+
+        let encoded = encode_byte(byte);
+
+        out.push(encoded);
+    }
+}
+
+fn serialize_range_mappings(sm: &SourceMap) -> Option<String> {
+    if sm.range_tokens.is_empty() {
+        return None;
+    }
+
+    let mut buf = Vec::new();
+    let mut prev_line = 0;
+
+    let mut idx_of_first_in_line = 0;
+    let mut indices = vec![];
+
+    for (idx, token) in sm.tokens().enumerate() {
+        if token.is_range() {
+            let num = idx - idx_of_first_in_line;
+
+            indices.push(num);
```

Comment:
Storing the indices first and converting to a bitvec later is a bit inefficient. This could just directly store the range mappings in a bitvec.